File: dataset/output/MISTRAL_codestral-latest/Task116_MISTRAL_codestral-latest.java
// Java
import java.io.IOException;

public class DropPrivileges {
    public static void main(String[] args) {
        try {
            Process p = Runtime.getRuntime().exec("id");
            p.waitFor();

            // Perform privileged operations here

            // Drop privileges
            String[] cmd = {"su", "-", "username", "-c", "id"};
            p = Runtime.getRuntime().exec(cmd);
            p.waitFor();
        } catch (IOException | InterruptedException e) {
            e.printStackTrace();
        }
    }
}

File: dataset/output/MISTRAL_codestral-latest/Task116_MISTRAL_codestral-latest.c
// C
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>

int main() {
    printf("Running as root
");
    uid_t uid = getuid();
    gid_t gid = getgid();

    // Perform privileged operations here

    // Drop privileges
    setgid(gid);
    setuid(uid);
    printf("Dropped privileges
");
    return 0;
}